From 8485f6ca1c7c348525951affadf46a5406db9651 Mon Sep 17 00:00:00 2001
From: Michael Ellerman <mpe@ellerman.id.au>
Date: Thu, 11 Jan 2018 14:37:30 +1100
Subject: [PATCH 1/2] rfi-flush: Move rfi_flush_fallback_area to end of paca
Patch-mainline: Submitted, 2018-01-09 LKML + linuxppc-dev
References: bsc#1075088

This was an oversight, it should be at the end of the paca, as that
reduces the risk of a kABI breakage. This also matches upstream.

Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
Reviewed-by: Torsten Duwe <duwe@suse.de>
---
 arch/powerpc/include/asm/paca.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/powerpc/include/asm/paca.h b/arch/powerpc/include/asm/paca.h
index 2d9d684..63a6b66 100644
--- a/arch/powerpc/include/asm/paca.h
+++ b/arch/powerpc/include/asm/paca.h
@@ -165,13 +165,13 @@ struct paca_struct {
 #endif
 #ifndef __GENKSYMS__
 #ifdef CONFIG_PPC_BOOK3S_64
-	void *rfi_flush_fallback_area;
 
 	/*
 	 * rfi fallback flush must be in its own cacheline to prevent
 	 * other paca data leaking into the L1d
 	 */
 	u64 exrfi[EX_SIZE] __aligned(0x80);
+	void *rfi_flush_fallback_area;
 	u64 l1d_flush_congruence;
 	u64 l1d_flush_sets;
 #endif
-- 
1.8.3.1

